bson/doc.go

```diff
@@ -19,24 +20,96 @@
+//
+//     3. An embedded struct field is marshalled as a subdocument. The key will be the lowercased name of the field's type.
+//
+//     4. A pointer field is marshalled as the underlying type if the pointer is non-nil. If the pointer is nil, it is
+//     marshalled as a BSON null value.
+//
+//     5. When unmarshalling, a field of type interface{} will follow the D/M type mappings listed above. BSON documents
+//     unmarshalled into an interface{} field will be unmarshalled as a D.
+//
+// The following struct tags can be used to configure behavior:
+//
+//     1. omitempty: If the omitempty struct tag is specified on a field, the field will not be marshalled if it is set to
+//     the zero value. By default, a struct field is only considered empty if the field's type implements the Zeroer
+//     interface and the IsZero method returns true. If the field's type does not implement Zeroer, the field will be
+//     marshalled as an embedded document.
+//
+//     2. minsize: If the minsize struct tag is specified on a field of type int64, uint, uint32, or uint64 and the value of
+//     the field can fit in a signed int32, the field will be serialized as a BSON int32 rather than a BSON int64. For other
+//     types, this tag is ignored.
+//
+//     3. truncate: If the truncate struct tag is specified on a field with a non-float64 numeric type, BSON doubles unmarshalled
+//     into that field will be trucated at the decimal point. For example, if 3.14 is unmarshalled into a field of type int,
+//     it will be unmarshalled as 3. If this tag is not specified, the decoder will throw an error if the value cannot be
+//     decoded without losing precision. For float64 or non-numeric types, this tag is ignored.
+//
+//     4. inline: If the inline struct tag is specified for a struct or map field, the field will be "flattened" when
+//     marshalling and "un-flattened" when unmarshalling. This means that all of the fields in that struct/map will be
+//     pulled up one level and will become top-level fields rather than being fields in a nested document. For example, if a
+//     map field named "Map" with value map[string]interface{}{"foo": "bar"} is inlined, the resulting document will be
+//     {"foo": "bar"} instead of {"map": {"foo": "bar"}}. There can only be one inlined map field in a struct. If there are
+//     duplicated fields in the resulting document when an inlined field is marshalled, an error will be returned. This tag
+//     can be used with fields that are pointers to structs. If an inlined pointer field is nil, it will not be marshalled.
+//     For fields that are not maps or structs, this tag is ignored.
+//
+// Marshalling and Unmarshalling
+//
+// Manually marshallhing and unmarshalling can be done with the Marshal and Unmarshal family of functions. To read or
+// write BSON from a non-slice source, an Encoder or Decoder with a bsonrw.ValueWriter or bsonrw.ValueReader must be
```

Comment:
I'm not sure about this last sentence.  Will it make sense to most people in this context?  Just looking in marshal.go, I can't find the functions to which this sentence applies.

Reply:
I took this from the previous docs, but I'd be surprised if this ever applies to anyone. I don't think it applies to any of those functions. To do this, you'd have to create an implementation of `ValueWriter` or `ValueReader` and then use those to create an `Encoder` or `Decoder`. Given all of those, I think it's best to remove these comments altogether.